sentry-spring/src/test/kotlin/io/sentry/spring/reactive/SentryReactiveWebFilterTest.kt

```diff
@@ -0,0 +1,62 @@
+package io.sentry.spring.reactive
+
+import com.nhaarman.mockitokotlin2.check
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.sentry.Breadcrumb
+import io.sentry.IHub
+import io.sentry.SentryOptions
+import javax.servlet.ServletRequestEvent
+import kotlin.test.Test
+import org.assertj.core.api.Assertions
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest
+import org.springframework.mock.web.server.MockServerWebExchange
+import org.springframework.web.server.WebFilterChain
+
+class SentryReactiveWebFilterTest {
+
+    private class Fixture {
+        val baseHub = mock<IHub>()
+        val hub = mock<IHub>()
+        val filter = SentryReactiveWebFilter(baseHub, SentryOptions())
+        val request = MockServerHttpRequest
+            .post("http://localhost:8080/some-uri")
+            .build()
+        val event = mock<ServletRequestEvent>()
+        val webExchange = MockServerWebExchange.from(request)
+        val chain = mock<WebFilterChain>()
+
+        init {
+            whenever(baseHub.clone()).thenReturn(hub)
+        }
+    }
+
+    private val fixture = Fixture()
+
+    @Test
+    fun `clones hub when request arrives`() {
+        fixture.filter.filter(fixture.webExchange, fixture.chain)
+
+        verify(fixture.baseHub).clone()
+        verify(fixture.hub).pushScope()
+    }
```

Comment:
should we not `pop` the scope once the request is over?

Reply:
I just added popScope, here in the last commit